test/export/StructureDefinition.LogicalExporter.test.ts

```diff
@@ -328,6 +328,82 @@ describe('LogicalExporter', () => {
     );
   });
 
+  it('should include added element having logical model as datatype when parent is Element', () => {
+    const logicalFoo = new Logical('Foo');
+    logicalFoo.parent = 'Element';
+    const addElementRuleBars = new AddElementRule('bars');
+    addElementRuleBars.min = 0;
+    addElementRuleBars.max = '1';
+    addElementRuleBars.types = [{ type: 'Bar' }];
+    addElementRuleBars.short = 'short of property bars';
+    logicalFoo.rules.push(addElementRuleBars);
+    doc.logicals.set(logicalFoo.name, logicalFoo);
+
+    const logicalBar = new Logical('Bar');
```

Comment:
To really test your logic, I think it's actually `Bar` that needs to have the parent set to `Element` -- because `Bar` is the type we are attempting to constrain (and get the `sdType` in the process).  If you set `logicalBar.parent` to `'Element'` here the test will fail.